datachannel.go

```diff
@@ -457,6 +460,12 @@ func (d *DataChannel) Close() error {
 		return nil
 	}
 
+	if d.readLoopActive != nil {
```

Comment:
What do you think of dropping the `nil` check or logging a `error` message? I would rather have a crash/error message over inconsistent behavior. 

Is closing a `nil` chan safe?

Reply:
I'm okay doing a check for `!d.api.settingEngine.detach.DataChannels` instead, which is when we have a read loop. What would you prefer?